### ISSUE TITLE
Asset caching

### ISSUE THREAD

**Issue by limzykenneth** [inlife/nexrender#834]
**Is your feature request related to a problem? Please describe.**
With [nexrender-action-cache](https://github.com/inlife/nexrender/tree/master/packages/nexrender-action-cache) it is possible to cache the project template (AE file) with a TTL when downloaded over a network. However this caching is not implemented for assets yet (as per #616).

**Describe the solution you'd like**
I'm moving more and more assets around my network when rendering so being able to cache assets as well instead of hitting the network all the time will be a great thing to have. We may also need to look into more cache invalidation and per asset cache bypass methods.

**Describe alternatives you've considered**
Since nexrender-action-cache is already there, I thought I build on it if possible.

**Additional context**
I'll try to look into the implementation in a bit.

**Comment by inlife** [comment]
Actually, that is another great idea. I have not thought of that. Will be looking forward to the implementation!

**Comment by inlife** [comment]
NVM, reading back the original issue I see I mentioned something like this, but already forgot about that 😄

**Comment by limzykenneth** [comment]
As mentioned in #835, I will probably be trying an alternative implementation of caching. Since I'm only focusing on caching network resources, specifically HTTP sources, I'll probably build it based off of the HTTP cache standards.

**Comment by inlife** [comment]
Fully support using HTTP cache standards or any existing standards for that matter. 👍  I think the URI standard, for example, fits very well with this library, and I don't even want to imagine some other custom solution that might have been added instead.

**Comment by limzykenneth** [comment]
I found a pretty ideal solution in the form of [make-fetch-happen](https://github.com/npm/make-fetch-happen) which is a `fetch` polyfill used and maintained by NPM that support HTTP caching semantics plus a bunch of really nice stuff such as automatic retry, proxy, and custom TLS certificates.

Since it is a fetch polyfill, we can just use it as a drop in replacement for node-fetch currently in use. Caching functionalities would need to be defined by passing an extra `params` property to the assets that is passed directly to make-fetch-happen but it also means it is completely backwards compatible.

I've tried it out in a simple test and it works perfectly without issue, given that the server has implementation for the caching related HTTP headers. If that looks good, I can go ahead with the implementation directly to nexrender-core.
